src/game/handlers/upStat.js

```diff
@@ -11,19 +12,22 @@ import { level } from '../core/level'
 import { reject } from './errors'
 
 import { capitalize } from './helpers'
-import { getStatCost, getCurrentStatPoints, statIds } from './statHelpers'
+import { statUpgradeCost, unspentStatPoints, statIds } from './statHelpers'
 
-function increaseStat (dao, char, statId) {
-  const amount = char[statId]
+function updateStat (dao, char, statId, changeAmount = 1) {
+  const currentStatScore = char[statId]
   const query = pick(['_id', 'updatedAt'], char)
+
   return dao.character.update(query, {
-    $set: { [statId]: amount + 1 },
+    $set: { [statId]: currentStatScore + changeAmount },
   })
 }
 
 export default function call (dao, provider, _, msg) {
-  const statName = msg.matches[1]
+  const statName = msg.matches[2]
   const statId = statIds[statName]
+  const changeType = msg.matches[1] === 'refund' ? 'refund' : 'upgrade'
```

Comment:
I'd prefer validating (error message) over assuming  'upgrade' for anything different than `refund`.